packages/vscode-pyright/src/extension.ts

```diff
@@ -115,6 +116,32 @@ export async function activate(context: ExtensionContext) {
 
     cancellationStrategy = new FileBasedCancellationStrategy();
     let serverOptions: ServerOptions | undefined = undefined;
+    let serverBaseOptions: Partial<Executable> = {
+        args: cancellationStrategy.getCommandLineArguments(),
+    };
+    switch (workspace.getConfiguration('basedpyright.transport').get('kind')) {
+        case 'ipc':
+            serverBaseOptions.transport = TransportKind.ipc;
+            break;
+        case 'socket':
+            const port = workspace.getConfiguration('basedpyright.transport').get('port');
+            if (port !== -1) {
+                serverBaseOptions.transport = {
+                    kind: TransportKind.socket,
+                    port: Number(port),
+                };
+                break;
+            }
+            serverBaseOptions.transport = TransportKind.socket;
+            break;
```

Comment:
`"socket"` doesn't seem to work in either `"useBundled"` (silently doesn't start, no error) or `"fromEnvironment"` (see error in the thread above)